runtime/libia2/include/ia2_internal.h

```diff
@@ -408,9 +434,40 @@ __attribute__((__noreturn__)) void ia2_reinit_stack_err(int i);
 #endif
 /* clang-format on */
 
+#ifdef IA2_NO_DEFS
+#define _IA2_INIT_RUNTIME(n)                                                   \
+  __attribute__((visibility("default"))) extern int ia2_n_pkeys_to_alloc;                                                \
+  __attribute__((visibility("default"))) extern __thread void *ia2_stackptr_0[]                    \
+      __attribute__((aligned(4096)));                                          \
+                                                                               \
+  REPEATB(n, declare_init_tls_fn, nop_macro);                                  \
+                                                                               \
+  /* Returns `&ia2_stackptr_N` given a pkru value for the Nth compartment. */  \
+  __attribute__((visibility("default"))) void **ia2_stackptr_for_pkru(uint32_t pkru) {                                \
+    REPEATB(n, return_stackptr_if_compartment,                                 \
+            return_stackptr_if_compartment);                                   \
+    return NULL;                                                               \
+  }                                                                            \
+                                                                               \
+  __attribute__((visibility("default"))) __attribute__((weak)) void init_stacks_and_setup_tls(void) {                 \
+    verify_tls_padding();                                                      \
+    COMPARTMENT_SAVE_AND_RESTORE(REPEATB(n, ALLOCATE_COMPARTMENT_STACK_AND_SETUP_TLS, nop_macro), n); \
+    /* allocate an unprotected stack for the untrusted compartment */          \
+    allocate_stack_0();                                     \
+  }                                                                            \
+                                                                               \
+  __attribute__((constructor)) static void ia2_init(void) {                    \
+    /* Set up global resources. */                                             \
+    ia2_set_up_tags(&ia2_n_pkeys_to_alloc);                                    \
+    /* Initialize stacks for the main thread/ */                               \
+    init_stacks_and_setup_tls();                                               \
+    REPEATB##n(setup_destructors_for_compartment, nop_macro);                  \
+    mark_init_finished();                                                      \
+  }
+#else
```

Comment:
@ayrtonm @fw-immunant Do either of you remember what this was about? We don't seem to be using `IA2_NO_DEFS` anywhere else, so I'm guessing it's not relevant anymore.